File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const newSet = new Set();
    for (let num of nums){
        if(newSet.has(num)){
            return true;
        } else {
            newSet.add(num);
        }
    }
    return false;
};

File: 0217-contains-duplicate/0217-contains-duplicate.java
import java.util.HashSet;

class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet <Integer> newSet = new HashSet<Integer>();
        for(int i=0; i<nums.length; i++){
            if(newSet.contains(nums[i])){
                return true;
            } else {
                newSet.add(nums[i]);
            }
        }
        return false;
    }
}